File: solutions/beecrowd/1218/1218.cpp
#include <cstdint>
#include <iostream>
#include <string>

int main() {
    int16_t m, f, c = 0;
    std::string s, l, cs, g;

    while (std::cin >> s) {
        m = 0;
        f = 0;

        getline(std::cin, l);  // it's a hack
        getline(std::cin, l);

        if (c) {
            std::cout << std::endl;
        }

        for (size_t i = 0; i < l.length(); i += 5) {
            if (s.at(0) == l.at(i) && s.at(1) == l.at(i + 1)) {
                if (l.at(i + 3) == 'M') {
                    m++;
                } else {
                    f++;
                }
            }
        }

        std::cout << "Caso " << ++c << ":" << std::endl;
        std::cout << "Pares Iguais: " << m + f << std::endl;
        std::cout << "F: " << f << std::endl;
        std::cout << "M: " << m << std::endl;
    }

    return 0;
}

File: solutions/beecrowd/1218/1218.py
import sys

input = sys.stdin.readline
output = sys.stdout.write

case = 0
while shoe_number_target := input().strip():
    if case:
        output('\n')
    case += 1

    shoes_female = 0
    shoes_male = 0

    shoes = iter(input().split())

    while shoe_number := next(shoes, False):
        shoe_gender = next(shoes)

        if shoe_number == shoe_number_target:
            if shoe_gender == 'M':
                shoes_male += 1
            else:
                shoes_female += 1

    output(f'Caso {case}:\n')
    output(f'Pares Iguais: {shoes_male + shoes_female}\n')
    output(f'F: {shoes_female}\n')
    output(f'M: {shoes_male}\n')
